frontend/src/main/components/RecommendationRequest/RecommendationRequestForm.js

```diff
@@ -0,0 +1,140 @@
+import { Button, Form, Row, Col } from "react-bootstrap";
+
+import { useForm } from "react-hook-form";
+import { useNavigate } from "react-router-dom";
+
+function RecommendationRequestForm({
+  initialContents,
+  submitAction,
+  buttonLabel = "Create",
+}) {
+  // Stryker disable all
+  const {
+    register,
+    formState: { errors },
+    handleSubmit,
+  } = useForm({ defaultValues: initialContents || {} });
+  // Stryker restore all
+
+  const navigate = useNavigate();
+
+  // For explanation, see: https://stackoverflow.com/questions/3143070/javascript-regex-iso-datetime
+  // Note that even this complex regex may still need some tweaks
+
+  // Stryker disable Regex
+
+  return (
+    <Form onSubmit={handleSubmit(submitAction)}>
+      <Row>
+        {initialContents && (
+          <Col>
+            <Form.Group className="mb-3">
+              <Form.Label htmlFor="id">Id</Form.Label>
+              <Form.Control
+                data-testid="RecommendationRequestForm-id"
+                id="id"
+                type="text"
+                {...register("id")}
+                value={initialContents.id}
+                disabled
+              />
+            </Form.Group>
+          </Col>
+        )}
+
+        <Col>
+          <Form.Group className="mb-3">
+            <Form.Label htmlFor="professorName">Professor Name</Form.Label>
+            <Form.Control
+              data-testid="RecommendationRequestForm-professorName"
+              id="professorName"
+              type="text"
+              isInvalid={Boolean(errors.professorName)}
+              {...register("professorName", {
+                required: true,
+              })}
+            />
+            <Form.Control.Feedback type="invalid">
+              {errors.professorName && "Professor Name is required. "}
+            </Form.Control.Feedback>
+          </Form.Group>
+        </Col>
```

Comment:
Instead of asking for professor name, there should be a dropdown component, with all of the Users in the users table that have the professor role.   Store the userId of that selected professor.